tiledb/sm/query/query_condition.cc

```diff
@@ -1558,12 +1603,19 @@ void QueryCondition::apply_ast_node_sparse(
     const tdb_unique_ptr<ASTNode>& node,
```

Comment:
Add template parameter for nullable here and use `if constexpr` to specialize the processing below. Caller will need to pick specialization based on the data.

Reply:
Done!